LibreNMS/Authentication/ActiveDirectoryAuthorizer.php

```diff
@@ -23,6 +23,7 @@ public function authenticate($credentials)
         $this->connect();
 
         if ($this->ldap_connection) {
+            $credentials['username'] = $this->getUsersAMAccountName($credentials['username']);
             // bind with sAMAccountName instead of full LDAP DN
             if (!empty($credentials['username']) && !empty($credentials['password']) && ldap_bind($this->ldap_connection, $credentials['username'] . '@' . Config::get('auth_ad_domain'), $credentials['password'])) {
```

Comment:
You should include `$credentials['username'] . '@' . Config::get('auth_ad_domain'` in your getUserSamAccountName method so it returns a ready to bind username.

Reply:
sAMAccountName can differ from UserPrincipalName.

For me test I set the attribut :
- sAMAccountName : dgds
- UserPrincipalName : dogus@lab.local

in LegacyUserProvider I call this method to set the username (in librenms db) to  sAMAccountName.

And in some method like userFilter we use this filter : 
` $user_filter = "(&(samaccountname=$username)(!(useraccountcontrol:1.2.840.113556.1.4.803:=2))";`
I don't know if samaccountname=username@domain works

Reply:
@dGs- 
```
$credentials['username'] = $this->getUsersAMAccountName($credentials['username']);
ldap_bind($this->ldap_connection, $credentials['username'] . '@' . Config::get('auth_ad_domain'), $credentials['password'])
```
These means the code always binds with the domain appended and it always calls `getUsersAMAccountName()`...  That is what I was saying two things changing the username 100% of the time, it should happen once.

Is there a reason you want the username in the db to be sAMAccountName?

Reply:
Yes because if we don't keep sAMAccountName as username in the db, it will create a new account for those who currently use active_directory mechanism, and they will loose theirs private dashboards.